cartography/intel/github/util.py

```diff
@@ -10,16 +13,42 @@
 
 import requests
 
+
 logger = logging.getLogger(__name__)
 # Connect and read timeouts of 60 seconds each; see https://requests.readthedocs.io/en/master/user/advanced/#timeouts
 _TIMEOUT = (60, 60)
+_GRAPHQL_RATE_LIMIT_REMAINING_THREASHOLD = 500
 
 
 class PaginatedGraphqlData(NamedTuple):
     nodes: List[Dict[str, Any]]
     edges: List[Dict[str, Any]]
 
 
+def handle_rate_limit_sleep(token: str) -> None:
+    '''
+    Check the remaining rate limit and sleep if remaining is below threshold
+    :param token: The Github API token as string.
+    '''
+    response = requests.get('https://api.github.com/rate_limit', headers={'Authorization': f"token {token}"})
```

Comment:
Rather than add another call, would it be possible to use the `rateLimit` object in the graphql response object itself? https://github.com/lyft/cartography/blob/f3f237cb9625f26205ead1da971d5fb7cf3127b0/cartography/intel/github/teams.py#L91-L96

This would let us save an API call for each page.

Reply:
That's why I tried originally, but if the graphql ratelimit is already fully consumed, then the query would not return a response. Calling the rest endpoint consumes a different ratelimit, `core` I believe. Test data shows that the limit for core is 60, but it's actually >5k for authenticated requests.

Reply:
From conversation we had offline, if we start the sync without knowing that the `remaining` quota is already 0, then the graphql response won't even return and the code will crash.